lib_camera/src/asm/rgb_to_gs16.S

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+
+// the cpu version
+/*
+void isp_rgb_to_greyscale16(int8_t * out_img, int8_t * img, size_t img_size)
+*/
+/*
+Sooo, for the coefficients:
+
```

Comment:
I deeply appreciate the comments giving hints for next HW or possible modifications, although it would be nice to add another comment that is more specific about this current implementation. 
for example say that this is doing the conversion: 0.299 ∙ Red + 0.587 ∙ Green + 0.114 ∙ Blue (NTSC formula)